source/client/renderer/GameRenderer.cpp

```diff
@@ -542,12 +593,142 @@ void GameRenderer::renderLevel(float f)
+	m_isometricZ = 0;
+}
+
+void GameRenderer::beginIsom(bool &oldDontRenderGui)
+{
+#ifdef CONTROLLABLE_ISOM
+	static float x = -1.0f;
+	oldDontRenderGui = false;
+	if (!m_bIsometric) {
+		x = -1.0f;
+		return;
+	}
+
+	if (m_isomStage > 10000)
+	{
+		m_bIsometric = false;
+		x = -1.0f;
+		return;
+	}
+
+	float deltaTime = 0.0f;
+
+	if (x < 0)
+		x = getTimeS();
+
+	deltaTime = getTimeS() - x;
+	x += deltaTime;
+
+	if (Keyboard::isKeyDown(m_pMinecraft->getOptions()->getKey(eKeyMappingIndex::KM_FORWARD))) {
+		m_isometricZ += deltaTime * 10;
+	}
+	if (Keyboard::isKeyDown(m_pMinecraft->getOptions()->getKey(eKeyMappingIndex::KM_BACKWARD))) {
+		m_isometricZ -= deltaTime * 10;
+	}
+	if (Keyboard::isKeyDown(m_pMinecraft->getOptions()->getKey(eKeyMappingIndex::KM_LEFT))) {
+		m_isometricX += deltaTime * 10;
+	}
+	if (Keyboard::isKeyDown(m_pMinecraft->getOptions()->getKey(eKeyMappingIndex::KM_RIGHT))) {
+		m_isometricX -= deltaTime * 10;
+	}
+	if (Keyboard::isKeyDown(m_pMinecraft->getOptions()->getKey(eKeyMappingIndex::KM_DESTROY))) {
+		m_isometricY += deltaTime * 10;
+	}
+	if (Keyboard::isKeyDown(m_pMinecraft->getOptions()->getKey(eKeyMappingIndex::KM_PLACE))) {
+		m_isometricY -= deltaTime * 10;
+	}
+
+	LOG_I("isom: %f %f %f", m_isometricX, m_isometricY, m_isometricZ);
+
+	return;
+#else
+	oldDontRenderGui = false;
+	if (!m_bIsometric)
+		return;
+#endif
+
+	const float isomSizeX = C_ISOM_TILES_PER_WIDTH(float(Minecraft::width) / float(Minecraft::height));
+	const float isomSizeY = C_ISOM_TILES_PER_HEIGHT;
+	const int isomWidth = int(C_MAX_CHUNKS_X * 16 / isomSizeX / 2) + 2;
+	const int isomHeight = int(C_MAX_CHUNKS_X * C_MAX_CHUNKS_Z / isomSizeY) + 2;
+	const int maxIsomStage = isomWidth * isomHeight;
+
+	if (m_isomStage >= maxIsomStage) {
+		m_isomStage = 0;
+		m_bIsometric = false;
+	}
+	else {
+		const float twoIsomSizeX = 2 * isomSizeX;
+		
+		float xInImg = float(m_isomStage % isomWidth - isomWidth / 2) * isomSizeX;
+		float zInImg = float(m_isomStage / isomWidth) * isomSizeY;
+
+		m_isometricX = 0.0f;
+		m_isometricZ = 0.0f;
+		m_isometricY = C_MAX_Y * 3 / 4; // TODO: fix magic value
+
+		// apply the displacement
+		m_isometricX += 2 * xInImg;
+		m_isometricZ -= 2 * xInImg;
+		m_isometricX += zInImg;
+		m_isometricZ += zInImg;
+
+		m_isomStage++;
+
+		oldDontRenderGui = m_pMinecraft->getOptions()->m_bDontRenderGui;
+		m_pMinecraft->getOptions()->m_bDontRenderGui = true;
+	}
+}
+
+void GameRenderer::endIsom(bool oldDontRenderGui)
+{
+	const float isomSizeX = C_ISOM_TILES_PER_WIDTH(float(Minecraft::width) / float(Minecraft::height));
+	const float isomSizeY = C_ISOM_TILES_PER_HEIGHT;
+	const int isomWidth = int(C_MAX_CHUNKS_X * 16 / isomSizeX / 2) + 2;
+	const int isomHeight = int(C_MAX_CHUNKS_X * C_MAX_CHUNKS_Z / isomSizeY) + 2;
+	const int maxIsomStage = isomWidth * isomHeight;
+
+	if (m_bIsometric) {
+		LOG_I("Saving %d of %d", m_isomStage, maxIsomStage);
+		std::string fn = "IsomShots\\" + std::to_string(m_isomStage) + ".png";
```

Comment:
windows detected!

Reply:
This is also temporary, and will be replaced.